platform/darwin/test/MGLExpressionJSTests.mm

```diff
@@ -0,0 +1,220 @@
+#import <XCTest/XCTest.h>
+
+#import "MGLTypes.h"
+#import "NSExpression+MGLPrivateAdditions.h"
+
+@interface MGLExpressionJSTests : XCTestCase
+
+@end
+
+@implementation MGLExpressionJSTests
+
+- (void)testAllJavascriptTests {
+    NSString *testRootPath = @"expression-tests";
+
+    NSBundle *bundle = [NSBundle bundleForClass:[self class]];
+    NSURL *rootURL = [[bundle bundleURL] URLByAppendingPathComponent:testRootPath];
+
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSDirectoryEnumerator *enumerator = [fileManager enumeratorAtURL:rootURL
+                                          includingPropertiesForKeys:@[NSURLIsDirectoryKey]
+                                                             options:0
+                                                        errorHandler:nil];
+    for (NSURL *fileURL in enumerator) {
+        if ([[fileURL lastPathComponent] isEqual:@"test.json"] && ![self shouldSkipTest:fileURL] ) {
+            NSData *data = [NSData dataWithContentsOfURL:fileURL];
+            NSError *error = nil;
+            NSDictionary *jsonDict = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+
+            if (!jsonDict && error) {
+                XCTFail(@"================> JSON parse failed with error: %@", error.localizedDescription);
+                continue;
+            }
+
+            try {
+                [self runTestsWithDictionary:jsonDict];
+            } catch (NSException *e) {
+                XCTFail(@"================> File at URL: %@ failed with reason: %@", fileURL, e.reason);
+                continue;
+            }
+        }
+    }
+}
+
+NSSet *testsToSkip = [NSSet setWithObjects:
+                      @"to-boolean",
+                      @"concat/arity-1",
```

Comment:
#11472 will make it possible to express a concatenation expression with an arity of 1 within an NSExpression. This is the case for `concat` but not, say, `+`, because `concat` takes a series of arguments, whereas `mgl_join:` would take a single argument containing all the string expressions to concatenate. By contrast, `add:to:` takes two arguments.

On the subject of arity, it’s worth pointing out that built-in NSExpression functions like `+` are strictly binary, such that `1 + 2 + 3` is interpreted as `(1 + 2) + 3`. Currently, `["+", 1, 2, 3]` is converted to `sum({1, 2, 3})`, which I think is reasonable but not a literal round trip.